src/guide/communicating_fees.md

```diff
@@ -0,0 +1,234 @@
+# Communicating fees
+
+In the LSP model, fees are involved when the user wants to receive a payment, but doesn't have sufficient receivable amount. This section provides recommendations on how to communicate these fees to a user.
+
+### Before receiving a payment
+When the user wants to receive a payment, a setup fee is paid when the resulting invoice would exceed the receivable amount.
+The setup fee is made up of two parts:
+- A minimum fee
+- A proportional fee based on the amount
+
+Before creating an invoice, the amount the user will want to receive is yet unknown. It is recommended to show the user a message consisting of the following information:
+
+> A setup fee of x% with a minimum of y sats will be applied for receiving more than z sats.
+
+You can construct this message as follows:
+
+<custom-tabs category="lang">
+<div slot="title">Rust</div>
+<section>
+
+```rust,ignore
+{{#include ../../snippets/rust/src/communicating_fees.rs:get-fee-info-before-receiving-payment}}
+```
+</section>
+
+<div slot="title">Swift</div>
+<section>
+
+```swift,ignore
+{{#include ../../snippets/swift/BreezSDKExamples/Sources/CommunicatingFees.swift:get-fee-info-before-receiving-payment}}
+```
+</section>
+
+<div slot="title">Kotlin</div>
+<section>
+
+```kotlin,ignore
+{{#include ../../snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/CommunicatingFees.kt:get-fee-info-before-receiving-payment}}
+```
+</section>
+
+<div slot="title">React Native</div>
+<section>
+
+```typescript
+{{#include ../../snippets/react-native/communicating_fees.ts:get-fee-info-before-receiving-payment}}
+```
+</section>
+
+<div slot="title">Dart</div>
+<section>
+
+```dart,ignore
+{{#include ../../snippets/dart_snippets/lib/communicating_fees.dart:get-fee-info-before-receiving-payment}}
+```
+</section>
+
+<div slot="title">Python</div>
+<section>
+
+```python,ignore
+{{#include ../../snippets/python/src/communicating_fees.py:get-fee-info-before-receiving-payment}}
+```
+</section>
+
+<div slot="title">Go</div>
+<section>
+
+```go,ignore
+{{#include ../../snippets/go/communicating_fees.go:get-fee-info-before-receiving-payment}}
+```
+</section>
+
+<div slot="title">C#</div>
+<section>
+
+```cs,ignore
+{{#include ../../snippets/csharp/CommunicatingFees.cs:get-fee-info-before-receiving-payment}}
+```
+</section>
+</custom-tabs>
+
+### When an invoice is created
+After calling `receive_payment`, you would typically show the recipient a screen containing a QR code with the invoice that the sender can scan.
+
+This is another point to show the user the opening fees applied to the invoice. At this point the amount the user wants to receive is known, so the message can be more concise:
+
+> A setup fee of x sats is applied to this invoice.
+
+The fiat amount can be included. In case of a mobile app it is recommended to communicate to the user that the app has to be run in the foreground in order to be able to receive the payment.
```

Comment:
Can we add this developer note here?
```suggestion
The fiat amount can also be included. In case of a mobile app it is recommended to communicate to the user that the app has to be run in the foreground in order to be able to receive the payment.
<div class="warning">
<h4>Developer note</h4>
Consider implementing the <a href="/notifications/getting_started.md">Notification Plugin</a> when using the Breez SDK in a mobile application. By registering a webhook the application can receive a payment notification to process the payment in the background.
</div>
```